docs/guides/Package-authoring-best-practices.md

```diff
@@ -0,0 +1,177 @@
+---
+title: Package authoring best practices
+description: A general guide of best practices for creating high quality NuGet packages.
+author: chgill-MSFT
+ms.author: chgill
+ms.date: 09/17/20
+ms.topic: conceptual
+---
+
+# Package authoring best practices
+
+This guidance is intended to give NuGet.org package authors a light-weight end to end reference for creating and publishing high quality packages. It will primarily focus on package-specific best practices such as metadata and packing. For more in depth suggestions for building high quality libraries, see the .NET [Open-source library guidance](https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/).
+
+## Types of recommendations
+
+Each article presents four types of recommendations: **Do**, **Consider**, **Avoid**, and **Do not**. The type of recommendation indicates how strongly it should be followed.
+
+You should almost always follow a **Do** recommendation. For example:
+
+✔️ DO include a short description for your package that describes what it's for.
+
+On the other hand, **Consider** recommendations should generally be followed, but there are legitimate exceptions to the rule and you shouldn't feel bad about not following the guidance:
+
+✔️ CONSIDER choosing a NuGet package name with a prefix that meets NuGet's prefix reservation [criteria](https://docs.microsoft.com/en-us/nuget/reference/id-prefix-reservation).
+
+**Avoid** recommendations mention things that are generally not a good idea, but breaking the rule sometimes makes sense:
+
+❌ AVOID NuGet package references that demand an exact version.
+
+And finally, **Do not** recommendations indicate something you should almost never do:
+
+❌ DO NOT use the `LicenseUrl` metadataproperty.
+
```

Comment:
Before frameworks, should we talk about how to pack? We want to push people to SDK-style packing instead of .nuspecs right?

https://docs.microsoft.com/en-us/dotnet/standard/library-guidance/nuget#create-a-nuget-package

Reply:
My original outline planned to have that at the end since chronologically you pack after you've already written the package. What do you think?

Reply:
I think **where** you put the information is the first thing I'd be doing. Whether that's as XML in a .csproj or a .nuspec.

Reply:
I don't understand what you mean by "you pack after you've already written the package". While in the development cycle pack happens at the end, after you write code, compile and debug, the customer needs a project file in order to do these things. If they start out with a non-SDK style project, then write code, when they come to pack and read these recommendations, if the last thing is says is "hey, it would have been easier if you used an SDK style project", that's not great. I think guiding people to SDK style projects early is better (and this applies to ALL our docs, which are currently far too heavy on non-SDK style projects and custom nuspec files, since they were written before SDK style projects and never updated. But now I'm really getting off-topic.

Also, I don't currently see any guidance on preferring SDK style projects in the current file.

I think it's useful as one of the first recommendations, because even though they're not intended to be in order of importance, culturally often things at top of lists are more "important". Although there's also a case to be made about the last thing read is more likely to stick. However, some people will start reading from the top and not make it all the way to the end, so there might be greater reach by putting it at the top.

Reply:
Remember pack is really split on PackageReference vs non-PackageReference. 

legacy vs sdk has a difference in how you enable the functionality, but that's something that can be fixed.